grails-app/services/com/unifina/service/BalanceService.groovy

```diff
@@ -13,18 +17,19 @@ import java.util.concurrent.ExecutionException
 
 class BalanceService {
 	protected Web3Balance web3
-
+	protected EthereumModuleOptions ethereumOptions;
```

Comment:
Should not use `EthereumModuleOptions` here. Shared functionality should be broken off to some utility if necessary.